sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/ResourceManagerTestBase.java

```diff
@@ -319,6 +321,36 @@ private void setAccessible(final Field field) {
         });
     }
 
+    /**
+     * Builds the manager with provided http pipeline and profile in general manner.
+     *
+     * @param manager the class of the manager
+     * @param httpPipeline the http pipeline
+     * @param profile the azure profile
+     * @param <T> the type of the manager
+     * @return the manager instance
+     * @throws RuntimeException when field cannot be found or set.
+     */
+    protected <T> T buildManager(Class<T> manager, HttpPipeline httpPipeline, AzureProfile profile) {
+        try {
+            Constructor<T> constructor = manager.getDeclaredConstructor(httpPipeline.getClass(), profile.getClass());
+            AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
+                constructor.setAccessible(true);
+                return null;
+            });
+            return constructor.newInstance(httpPipeline, profile);
+
+        } catch (NoSuchMethodException ex) {
+            throw logger.logExceptionAsError(new RuntimeException(ex));
+        } catch (IllegalAccessException ex) {
+            throw logger.logExceptionAsError(new RuntimeException(ex));
+        } catch (InstantiationException ex) {
+            throw logger.logExceptionAsError(new RuntimeException(ex));
+        } catch (InvocationTargetException ex) {
```

Comment:
nit, java should be able to use `(NoSuchMethodException | IllegalAccessException | InstantiationException | InvocationTargetException  ex)`